### PR TITLE
infoschema: try on each PD member until one succeeds or all fail

### LINKED ISSUE #35268: TiDB should retry on every PD client URL until success or failure
### 1. Minimal reproduce step (Required)

1. Start a cluster with 3 PD instances.
2. Kill the PD server that is first in the list of members returned by the `pd.GetMemebers` and do not reboot it.
3. Try `show config` SQL on TiDB.

### 2. What did you expect to see? (Required)

TiDB executes the SQL successfully.

### 3. What did you see instead (Required)

origin_img_v2_e1106400-fe1b-44bc-8fa8-9355c7b2bf2g

### Root case

`GetPDServerInfo` should retry on every PD client URL until success or failure. In this case, if the first member is the PD server that crashes, it will never work until the PD server reboots.

https://github.com/pingcap/tidb/blob/7500d79a0af49daecbfd87a17e8eef391f9e3792/infoschema/tables.go#L1658-L1669

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: close #35268.

### What is changed and how it works?

`GetPDServerInfo` will try on each PD member until one succeeds or all fail.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

Before:

image

After:

image

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tiancaiamao — COMMENTED** [review]
Some PD server's information might be missing, and there is no warning about that?

**Comment by JmPotato** [comment]
> Some PD server's information might be missing, and there is no warning about that?

@tiancaiamao I added some warning logs, PTAL again, thx.

File: infoschema/tables.go [hunk 1]
```diff
@@ -1654,18 +1656,28 @@ func GetPDServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
...
+		errs      = make([]error, 0, memberNum)
+	)
+	// Try on each member until one succeeds or all fail.
 	for _, addr := range members {
 		// Get PD version, git_hash
 		url := fmt.Sprintf("%s://%s%s", util.InternalHTTPSchema(), addr, pdapi.Status)
 		req, err := http.NewRequest(http.MethodGet, url, nil)
 		if err != nil {
-			return nil, errors.Trace(err)
+			logutil.BgLogger().Warn("create pd server info request error", zap.String("url", url), zap.Error(err))
```
Comment (tiancaiamao) [inline infoschema/tables.go:1671]: You can try this `ctx.StmtCtx.AppendWarning(err)` to add a SQL warning.
The user may not check the tidb log.

Comment (JmPotato) [reply infoschema/tables.go:1672]: Done. PTAL.

Comment (JmPotato) [reply infoschema/tables.go:1672]: image

**Review by tiancaiamao — APPROVED** [review]

**Review by HunDunDM — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Review by Yisaer — COMMENTED** [review]
/merge

**Pull request merged**